op-node,op-service: Make L1 cache size configurable

sebastianst opened the pull request:
**Description**

Adds a new optional op-node flag `--l1.cache-size` to configure the L1 cache size. If not set (=0), a default of 2/3 the sequencing window is used.

Also removes the default limit of 1000, which was too low to hold more than ~3h of L1 data. 

A sanity check still happens that the cache size is not larger than 1 mio.

**Tests**

The op-node L1 rpc setup is untested and I don't intend to change this with this small improvement.

**Additional context**

This was causing problems whenever a duration of 3h has passed since a latest batch was posted, which then caused derivation of a new batch to fetch all L1 data again.
This particularly impacted chains that have a usual channel duration longer than 3h, or when chains experienced a safe head stall >3h.

**Metadata**

Fixes #13409.

protolambda commented on op-node/flags/flags.go:
```diff
@@ -179,6 +179,13 @@ var (
 		Value:    20,
 		Category: L1RPCCategory,
 	}
+	L1CacheSize = &cli.IntFlag{
```
There is `cli.Uint64Flag`, to make an unsigned int flag. That avoids the `< 0` check in the code later on.

sebastianst replied:
changed to `Uint`

protolambda commented on op-node/service.go:
```diff
@@ -160,6 +160,7 @@ func NewL1EndpointConfig(ctx *cli.Context) *node.L1EndpointConfig {
 		BatchSize:        ctx.Int(flags.L1RPCMaxBatchSize.Name),
 		HttpPollInterval: ctx.Duration(flags.L1HTTPPollInterval.Name),
 		MaxConcurrency:   ctx.Int(flags.L1RPCMaxConcurrency.Name),
+		CacheSize:        ctx.Int(flags.L1CacheSize.Name),
```
See flag comment about int type.

protolambda commented on op-node/node/client.go:
```diff
@@ -146,14 +155,20 @@ func (cfg *L1EndpointConfig) Setup(ctx context.Context, log log.Logger, rollupCf
-	return l1Node, rpcCfg, nil
+
+	var l1Cfg *sources.L1ClientConfig
+	if cfg.CacheSize > 0 {
```
Can we add a sanity-cap, like `if cfg.CacheSize > 10_000_000 { return nil, nil, fmt.Errorf("cache size is dangerously large") }`

sebastianst replied:
Added this check to `L1EndpointConfig.Check()` with a limit of 1 mio.

protolambda reviewed (approved):
LGTM. Just a few nits about types and sanity cap

sebastianst committed:
op-node,op-service: Make L1 cache size configurable

Also removes the default limit of 1000, which was too low to hold more
than ~3h of L1 data. This was causing problems whenever a duration of 3h
has passed since a latest batch was posted, which then caused derivation
of a new batch to fetch all L1 data again.
This particularly impacted chains that have a usual channel duration
longer than 3h, or when chains experienced a safe head stall >3h.

Fixes #13409.
